connector_test.go

```diff
@@ -123,37 +117,40 @@ var volumeConfig = `
 }
 `
 
-func TestSimpleVolume(t *testing.T) {
-	logger := log.NewTestLogger(t)
+func bindMountHelper(t *testing.T, options string) {
+	t.Helper()
 	fileContent, err := os.ReadFile("./tests/volume/test_file.txt")
 	assert.NoError(t, err)
 
-	connector, _ := getConnector(t, volumeConfig)
-	cwd, err := os.Getwd()
-	assert.NoError(t, err)
-	// disable selinux on the test folder in order to make the file readable from within the container
-	cmd := exec.Command("chcon", "-Rt", "svirt_sandbox_file_t", fmt.Sprintf("%s/tests/volume", cwd)) //nolint:gosec
-	err = cmd.Run()
-	if err != nil {
-		logger.Warningf("failed to set SELinux permissions on folder, chcon error: %s, this may cause test failure if SELinux is enabled.", err.Error())
-	}
+	connector, _ := getConnector(t, fmt.Sprintf(volumeConfig, options))
 
 	container, err := connector.Deploy(
 		context.Background(),
 		"quay.io/arcalot/podman-deployer-test-helper:0.1.0")
 	assert.NoError(t, err)
 
-	var containerInput = []byte("volume\n")
-	_, err = container.Write(containerInput)
-	assert.NoError(t, err)
+	t.Cleanup(func() { assert.NoError(t, container.Close()) })
+
+	assert.NoErrorR[int](t)(container.Write([]byte("volume\n")))
+
 	// Note: If it ends up with length zero buffer, restarting the VM may help:
 	// https://stackoverflow.com/questions/71977532/podman-mount-host-volume-return-error-statfs-no-such-file-or-directory-in-ma
 	readBuffer := readOutputUntil(t, container, string(fileContent))
 	assert.GreaterThan(t, len(readBuffer), 0)
+}
 
-	t.Cleanup(func() {
-		assert.NoError(t, container.Close())
-	})
+func TestBindMount(t *testing.T) {
+	scenarios := map[string]string{
+		"No options": "",
+		"Private":    ":Z",
+		"Shared":     ":z",
+		"ReadOnly":   ":ro",
+		"Multiple":   ":z,ro,noexec",
+	}
+	for name, s := range scenarios {
+		options := s
```

Comment:
The variable `s` has a scope ever so slightly larger than the body of the loop, whereas the scope of `options` is limited to the body of the loop.  This means that in the closure provided to the `t.Run()` function invocation, the value of `options` is fixed (because, on each iteration of the loop, a _new_ instance of `options` is created).  Prior to Go v1.22, if we were to reference the loop variable (e.g., `s`) inside that closure, its value would change as the loop iterates, and there is no way to guess which value of `s` would be used (most likely, it would be the value from the last iteration, because the loop would probably complete before any of the `t.Run()` invocations actually start).

So, "localizing" the value of `s` by instantiating a scoped instance of `options` protects us...until we get to `v1.22`, at which point this idiom will still work correctly.

Reply:
Ah: yeah, I guess I remember that from the parallelism discussions. It's incredibly weird, and will always look weird. (Sigh.)